pkg/cloud/metadata/lingjun_test.go

```diff
@@ -0,0 +1,220 @@
+package metadata
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// LingjunTestCases contains multiple test cases for Lingjun metadata
+var LingjunTestCases = map[string]string{
+	"standard": `{
+	"RegionId": "cn-hangzhou",
+	"ZoneId": "cn-hangzhou-a",
+	"NodeId": "i-1234567890abcdef0",
+	"InstanceType": "ecs.g6.large",
+	"EniMac": "00:11:22:33:44:55",
+	"AckNicName": "eth1",
+	"NimitzNetworkMode": "VPC",
+	"NimitzInterface": ["eth0", "eth1"]
+}`,
+	"minimal": `{
+	"RegionId": "cn-beijing",
+	"ZoneId": "cn-beijing-c",
+	"NodeId": "i-0987654321fedcba0",
+	"InstanceType": "ecs.c5.large"
+}`,
+	"extended": `{
+	"RegionId": "cn-shanghai",
+	"ZoneId": "cn-shanghai-b",
+	"NodeId": "i-1122334455aabbccdd",
+	"InstanceType": "ecs.r5.xlarge",
+	"EniMac": "aa:bb:cc:dd:ee:ff",
+	"AckNicName": "eth0",
+	"NimitzNetworkMode": "ENI",
+	"NimitzInterface": ["eth0", "eth1", "eth2"]
+}`,
+}
+
+func TestNewLingJunMetadata(t *testing.T) {
+	// Create a temporary directory for test files
+	tempDir := t.TempDir()
+	originalConfigFile := LingjunConfigFile
+
+	t.Run("successful parsing - standard", func(t *testing.T) {
+		// Setup test file
+		LingjunConfigFile = filepath.Join(tempDir, "lingjun_config_standard")
+		defer func() {
+			LingjunConfigFile = originalConfigFile
+		}()
```

Comment:
How about just pass `LingjunConfigFile` into `NewLingJunMetadata`? Replacing global variable should only be used as last resort, as it prevents parallel test.